### PR TITLE
Adds stack.

### PR DESCRIPTION
**Pull request by MicahZoltu**
I have had this sitting in my repo for a while.  No applied usage of it yet, but I would like feedback on it and if it is seen as being useful we can leverage it (not sure where exactly yet).

If there is no desire to have it merged into augur-core, I'll probably leave this branch up for a bit until I find a better home.

### CODE REVIEW

File: src/libraries/stack.se [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    return(removedValue)
+
+# should only be used with call = delegate
+def peek(key):
+    index = self.stacks[key].head
+    if (index == 0):
+        ~invalid()
+    return(self.stacks[key].nodes[index].item)
+
+def isEmpty():
```
Comment (joeykrug) [inline src/libraries/stack.se]: This is missing a `key` param

**Comment by joeykrug** [comment]
Also I think 

```
data stacks[](items[], head)

def push(key, item):
	self.stacks[key].head += 1
	index = self.stacks[key].head
	self.stacks[key].items[index] = item
	return(1)

def pop(key, item):
	index = self.stacks[key].head
	if(index == 0):
		~invalid()
	self.stacks[key].head = self.stacks[key].head - 1
	removedValue = self.stacks[key].items[index]
	self.stacks[key].items[index] = 0
	return(removedValue)

def peek(key):
	index = self.stacks[key].head
	if(index == 0):
		~invalid()
	return(self.stacks[key].items[index])

def isEmpty(key):
	return(self.stacks[key].head == 0)
```

Does the same thing at ~half or so the cost

**Comment by MicahZoltu** [comment]
Ah, heh.  When I wrote this originally it was a doubly linked list that allowed for removing from the middle.  I simplified it over the course of several iterations and didn't realize I had reached the point where one could just use an array.  Will update to your version.

**Comment by MicahZoltu** [comment]
Applied changes, ready for re-review.

**Pull request merged**